src/main/kotlin/open/tresorier/dao/jooq/pgsql/PgOperationDao.kt

```diff
@@ -106,15 +117,26 @@ class PgOperationDao(val configuration: Configuration) : IOperationDao {
         return operationList
     }
 
-    override fun findByBudget(budget: Budget): List<Operation> {
-        val jooqOperationList = this.query
-            .select()
-            .from(OPERATION)
-            .join(ACCOUNT).on(OPERATION.ACCOUNT_ID.eq(ACCOUNT.ID))
-            .where(ACCOUNT.BUDGET_ID.eq(budget.id))
-            .orderBy(OPERATION.MONTH.desc(), OPERATION.DAY.desc(), OPERATION.ORDER_IN_DAY.desc())
-            .fetch().into(OPERATION)
-
+    override fun findByBudget(budget: Budget, category: Category?): List<Operation> {
+        var jooqOperationList: org.jooq.Result<OperationRecord>
+        if (category != null) {
+            jooqOperationList = this.query
+                .select()
+                .from(OPERATION)
+                .join(ACCOUNT).on(OPERATION.ACCOUNT_ID.eq(ACCOUNT.ID))
+                .where(ACCOUNT.BUDGET_ID.eq(budget.id))
+                .and(OPERATION.CATEGORY_ID.eq(category.id))
+                .orderBy(OPERATION.MONTH.desc(), OPERATION.DAY.desc(), OPERATION.ORDER_IN_DAY.desc())
+                .fetch().into(OPERATION)
```

Comment:
you can add a if without having to write the query twice. see exemple in pgAllocationDao l.67